ui/permissions.py

```diff
@@ -68,4 +204,7 @@ def has_permission(self, request, view):
             uuid.UUID(collection_key)
         except ValueError:
             raise ValidationError('wrong UUID format for {}'.format(collection_key))
-        return Collection.objects.filter(owner=request.user, key=collection_key).exists()
+        collection = Collection.objects.filter(key=collection_key)
+        if collection:
+            return has_admin_permission(collection.first(), request)
+        return False
```

Comment:
`has_admin_permission` won't be run if the queryset result is empty:
```
matches = Collection.objects.filter(title='hfhhf') . # Returns empty <QuerySet []>
if matches:
    print('got a queryset containing 1+ collections, will run has_admin_permission() and collection.first()')
    has_admin_permission(collection.first(), request)
else:
    print('No collections in queryset, not running has_admin_permission() or collection.first()')
```
`> No collections in queryset, not running has_admin_permission() or collection.first()`